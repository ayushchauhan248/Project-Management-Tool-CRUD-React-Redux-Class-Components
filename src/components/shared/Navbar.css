#nav{
    display: flex;
    background-color: rgb(80, 73, 119);
    color: white;
    height: 50px;
    overflow: hidden;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 0;
    }
#left{
display: flex;
}
#brand{
    font-size: xx-large;
    padding-left: 8px;
    padding-top: 3px;
    font-family: 'Times New Roman', Times, serif
}
#img{
    margin-left: 20px;
    padding: 5px;
    height: 40px;
    width: 40px;
}
#icon{    
    height: 32px;
    width: 32px;
}
#llnk {
    text-decoration: none;
    color: white;
    font-size: large;
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
    border-bottom: 1px solid white;
}

#llnk:hover{
    background-color: rgb(80, 73, 119);
}

#iconDiv{
    top: 10px;
    position: absolute;
    right: 120px;
    cursor: pointer;
}

#buttonDiv{
    top:10px;
    position: absolute;
    right: 25px;
    cursor: pointer;
}
#username-div{
    font-size: large;
    font-weight: 600;
    top:12px;
    position: absolute;
    right: 160px;
}

#myProfile{
    color: white;
}