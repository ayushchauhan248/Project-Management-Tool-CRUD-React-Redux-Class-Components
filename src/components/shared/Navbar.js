import "./Navbar.css";
import { CgProfile } from "react-icons/cg";
import React, { Component } from "react";
import Button from "@mui/material/Button";
import { Link } from "react-router-dom";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../../theme";
import axios from "axios";
class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
    };
  }

  componentDidMount = async () => {
    const token = localStorage.getItem("AUTH_TOKEN");
    const url = "http://localhost:4000/api/users/me";
    try {
      const response = await axios.get(url, {
        headers: { "Content-Type": "application/json", Authorization: token },
      });
      this.setState(response.data);
    } catch (error) {
      console.log("Not able to fetch data");
    }
  };
  logout = () => {
    localStorage.removeItem("AUTH_TOKEN");
    window.location.href = "/login";
  };
  render() {
    return (
      <div id="nav">
        <div id="left">
          <img
            id="img"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQLEMlXG0H88o88hkRIK5vuAgYO57TCcGzczg&usqp=CAU"
            alt="LOGO"
          />
          <span id="brand">PROMAN</span>
        </div>
        <div id="username-div">{this.state.name}</div>
        <div id="iconDiv">
          <Link to="/myProfile" id="myProfile">
            <CgProfile id="icon"></CgProfile>
          </Link>
        </div>
        <ThemeProvider theme={theme}>
          <div id="buttonDiv">
            <Button
              size="small"
              variant="contained"
              color="primary"
              onClick={this.logout}
            >
              Logout
            </Button>
          </div>
        </ThemeProvider>
      </div>
    );
  }
}
export default Navbar;
