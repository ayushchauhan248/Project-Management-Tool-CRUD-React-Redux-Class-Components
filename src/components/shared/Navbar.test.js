import React from "react";
import Navbar from "./Navbar";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

it("should render snapshot of navbar", () => {
  const component = shallow(<Navbar />);
  console.log(component);
  expect(component).toMatchSnapshot();
});
