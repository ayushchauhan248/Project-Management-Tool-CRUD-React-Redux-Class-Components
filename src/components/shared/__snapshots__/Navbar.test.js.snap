// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render snapshot of navbar 1`] = `
<div
  id="nav"
>
  <div
    id="left"
  >
    <img
      alt="LOGO"
      id="img"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQLEMlXG0H88o88hkRIK5vuAgYO57TCcGzczg&usqp=CAU"
    />
    <span
      id="brand"
    >
      PROMAN
    </span>
  </div>
  <div
    id="iconDiv"
  >
    <Link
      id="myProfile"
      to="/myProfile"
    >
      <CgProfile
        id="icon"
      />
    </Link>
  </div>
</div>
`;
